### PR TITLE
Clarify documentation of SKIP_VERIFY

### LINKED ISSUE #12117: I don't think the "SKIP_VERIFY" is working for mailer settings
- Gitea version (or commit ref): 1.12.1 built with GNU Make 4.1, go1.14.4 : bindata, sqlite, sqlite_unlock_notify
- Git version: 2.27.0
- Operating system: Windows Server 2019
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:

## Description
I don't think the "SKIP_VERIFY" is working. Because when I use this setup. I get an error when sending a test email.
"gomail: could not send email 1: StartTLS: x509: certificate signed by unknown authority"

```
[mailer]
ENABLED = true
HOST    = mail.domain.com:587
FROM    = "GiteaServer"<email@arh-us.com>
USER    = email@arh-us.com
PASSWD  = password
IS_TLS_ENABLED = false
SKIP_VERIFY = 
MAILER_TYPE = smtp
```
...

### PR DESCRIPTION
**Pull request by cipherboy**
The documentation clearly documents the empty value as the default,
however at least one user reported this as being unclear. Mark values
explicitly so it is clear what values it can take. This clarifies that
an empty value in fact leaves certificate verification enabled, whereas
it has to be explicitly set to true to disable certificate verification.

Resolves: #12117

`Signed-off-by: Alexander Scheel <alexander.m.scheel@gmail.com>`

### CODE REVIEW

**Comment by silverwind** [comment]
Please also sync to [here](https://github.com/go-gitea/gitea/blob/6a62884782b29653bcd4462188e263018e5e21e8/custom/conf/app.example.ini#L646).

**Comment by cipherboy** [comment]
ACK, done. Thanks @silverwind.

File: docs/content/doc/advanced/config-cheat-sheet.en-us.md [hunk 1]
```diff
@@ -408,7 +408,8 @@ set name for unique queues. Individual queues will default to
 - `USER`: **\<empty\>**: Username of mailing user (usually the sender's e-mail address).
 - `PASSWD`: **\<empty\>**: Password of mailing user.  Use \`your password\` for quoting if you use special characters in the password.
    - Please note: authentication is only supported when the SMTP server communication is encrypted with TLS (this can be via `STARTTLS`) or `HOST=localhost`. See [Email Setup]({{< relref "doc/usage/email-setup.en-us.md" >}}) for more information.
-- `SKIP_VERIFY`: **\<empty\>**: Do not verify the self-signed certificates.
+- `SKIP_VERIFY`: **\<empty\>**: Whether or not to skip verification of certificates; empty or `false` to enable verification; `true` to disable verification.
```
Comment (mrsdizzie) [inline docs/content/doc/advanced/config-cheat-sheet.en-us.md]:
```suggestion
- `SKIP_VERIFY`: **false**: Whether or not to skip verification of certificates; `true` to disable verification.
```

File: custom/conf/app.example.ini [hunk 2]
```diff
@@ -643,7 +643,7 @@ HOST =
 DISABLE_HELO =
 ; Custom hostname for HELO operation, if no value is provided, one is retrieved from system.
 HELO_HOSTNAME =
-; Do not verify the certificate of the server. Only use this for self-signed certificates
+; !!Unsafe!! Whether or not to skip verification of certificates; empty or `false` to enable verification; `true` to disable verification. Consider adding the target certificate into the system trust store instead.
 SKIP_VERIFY =
```
Comment (mrsdizzie) [inline custom/conf/app.example.ini]:
```suggestion
SKIP_VERIFY = false
```

File: custom/conf/app.example.ini [hunk 3]
```diff
@@ -643,7 +643,7 @@ HOST =
 DISABLE_HELO =
 ; Custom hostname for HELO operation, if no value is provided, one is retrieved from system.
 HELO_HOSTNAME =
-; Do not verify the certificate of the server. Only use this for self-signed certificates
+; !!Unsafe!! Whether or not to skip verification of certificates; empty or `false` to enable verification; `true` to disable verification. Consider adding the target certificate into the system trust store instead.
```
Comment (mrsdizzie) [inline custom/conf/app.example.ini]:
```suggestion
; Whether or not to skip verification of certificates; `true` to disable verification. This option is unsafe. Consider adding the certificate to the system trust store instead.
```

**Review by mrsdizzie — CHANGES_REQUESTED** [review]
I think we should only use true/false for boolean examples and not mention empty. While empty will always evaluate to false, having it as a 3rd option seems to be what caused the original confusion this is meant to fix. Showing false as the default makes that more clear. Also slight change to comment just so it is the same between both files

**Comment by cipherboy** [comment]
Sorry, was out of internet range. Updated!

**Review by mrsdizzie — APPROVED** [review]

**Review by silverwind — APPROVED** [review]

**Pull request merged**